### ISSUE TITLE
Cannot mount ISO files from vagrant shared folder

### ISSUE THREAD

**Issue by ansiblejunky** [hashicorp/vagrant#9657]
### Vagrant version
Vagrant 2.0.3
No vagrant plugins installed

VirtualBox 5.1.18

### Host operating system
Mac OS X 10.12.6

### Guest operating system
Oracle Linux 6.9 (and others)

### Vagrantfile
```ruby
  config.vm.define "ol6-latest" do |config|
    config.vm.box = "ol6-latest"
    config.vm.box_url = "http://yum.oracle.com/boxes/oraclelinux/latest/ol6-latest.box"
  end
```

### Debug output
The following commands are performed on the machine and it results in error.
```bash
vagrant ssh ol6-latest
sudo -i
mkdir /media/iso
mount -t iso9660 -o loop,ro /vagrant/iso/rhel-server-6.9-x86_64-dvd.iso /media/iso

mount: wrong fs type, bad option, bad superblock on /dev/loop0,
       missing codepage or helper program, or other error
       In some cases useful info is found in syslog - try
       dmesg | tail  or so
```

### Expected behavior
ISO is mounted to /media/iso without errors

### Actual behavior
OS complains about the fs type or something. The strange thing here is that I can mount the same ISO on a different vagrant machine (Oracle Linux 5).  Additionally, when I create a new vagrant box (using packer) the mount command works! So there seems to be something with the vagrant machine that was created by Oracle.  Or, perhaps there is an issue with virtualbox guest additions? I truly do not know the root cause so I am posting this issue here as a start and hope for some good help.

Also note that if I copy the ISO to /tmp within the vagrant machine and then attempt to mount it, it works perfectly. So the issue seems to be something to do with the shared folder /vagrant .  The shared folder points to the local project folder on my Mac. Normal folder on my Mac file system.

### Steps to reproduce
1.  vagrant up (using vagrantfile listed above)
2. vagrant ssh
3. sudo to root
4. mount iso

**Comment by chrisroberts** [comment]
Hi there,

The issue you are encountering is not caused by Vagrant itself but rather the tools Vagrant utilizes (in this case VirtualBox). This is likely a limitation of vboxfs which is why it works when you copy it locally into the guest. If it works on other VirtualBox guests I would suggest checking the guest tools version on each and ensure they match. Outside of that, you may just have to copy the file into the guest prior to mounting.

Cheers!

**Comment by ansiblejunky** [comment]
From my research it appears the issue must rest with the machine itself and what packages or kernel is installed. I tested with different vbox guest additions and that is not the problem. Mounting the ISO works nicely on the same guest additions in a different machine. I tested with the same vbox version, same vagrant version, same vbox guest additions, and mounting the ISO on one machine works and another does not. 

Perhaps there is some package on my Oracle Linux that is required in order to support mounting this ISO from the vboxfs file system?
